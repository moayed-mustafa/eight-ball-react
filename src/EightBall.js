import React, { useState } from 'react'
import Eightball from './Eightball.css'


const EightBall = ({ answers }) => {

    const eve = () => {
        // here is where the state is changed
        let rand = Math.floor(Math.random() * answers.length)
        setAnswer(answers[rand])
        if (answers[rand].color == 'green')  setPositive(positive + 1)
        else if (answers[rand].color == 'red') setNegative(negative + 1)
        else setunDecisive(undecisive + 1)




    }
    const reset = () => {
        setAnswer({ msg: 'Think of a question', color: "black" })
        setPositive(0)
        setNegative(0)
        setunDecisive(0)
   }
    const [answer, setAnswer] = useState({msg:'Think of a question', color:"black"})
    const [positive, setPositive] = useState(0)
    const [negative, setNegative] = useState(0)
    const [undecisive, setunDecisive] = useState(0)
    return (
        <>
        <div className='eight-ball'>
            <div onClick={eve}className='eight-ball-circle' id={answer.color}>
                <p >{answer.msg}</p>
            </div>
            </div>
            <div>
        <button onClick={reset}>Reset</button>
                <p >positive: {positive}</p>
        <p >negative : {negative}</p>
        <p >undecisive : {undecisive}</p>

            </div>
            </>
    )

}

export default EightBall