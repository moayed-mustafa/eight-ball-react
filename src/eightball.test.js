import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import EightBall from './EightBall'
describe('smoke test', () => {
    test('EightBall renders without crashing', () => {
        render(<EightBall />)
    })

})
describe('snaphsot test test', () => {
    it('should match snapshot', () => {
        const { asFragment } = render(<EightBall />)
        expect(asFragment()).toMatchSnapshot()
    })

})
describe('matching elements test', () => {
    test('element exists', () => {
        const { getByText } = render(<EightBall />)
        expect(getByText("Reset", { exact: false })).toBeInTheDocument()
    })
})

describe('test events', () => {
    test('element reacts to event', () => {
        const { getByTestId, getByText } = render(<EightBall />)
        let circle = getByTestId("circle")
        let pos = getByText('positive: 0')
        expect(circle).toBeInTheDocument()
        //todo the click event is not working
        // fireEvent.click(circle)
        // expect(pos).toHaveTextContent('1')

    })
})

