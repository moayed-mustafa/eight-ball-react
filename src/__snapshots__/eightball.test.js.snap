// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snaphsot test test should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="eight-ball"
  >
    <div
      class="eight-ball-circle"
      data-testid="circle"
      id="black"
    >
      <p>
        Think of a question
      </p>
    </div>
  </div>
  <div>
    <button>
      Reset
    </button>
    <p>
      positive: 0
    </p>
    <p>
      negative : 0
    </p>
    <p>
      undecisive : 0
    </p>
  </div>
</DocumentFragment>
`;
